test/e2e/network/netpol/probe.go

```diff
@@ -64,6 +74,12 @@ func ProbePodToPodConnectivity(k8s *kubeManager, model *Model, testCase *TestCas
 	for i := 0; i < size; i++ {
 		result := <-results
 		job := result.Job
+		if job.PodFrom == nil {
```

Comment:
I don't these checks are necessary, if you actually check all the error conditions when constructing the model.

I'd recommend doing that -- make sure that the model is always in a valid state, so that these kinds of checks don't have to be performed by all clients.

Reply:
done